test/tests/deepkitType.test.ts

```diff
@@ -0,0 +1,99 @@
+import { classToPlain, t, plainToClass } from '@deepkit/type';
+import { getModelForClass, prop } from '../../src/typegoose';
+
+enum Group {
+  confidential = 'confidential',
+  public = 'public',
+}
+
+class Account {
+  @t.mongoId.group(Group.public)
+  public _id: string;
+
+  @t.group(Group.public)
+  public __v: number;
+
+  @t.group(Group.public)
+  @prop()
+  public email: string;
+
+  @t.group(Group.confidential)
+  @prop()
+  public confidentialProperty?: string;
+}
+
+const AccountModel = getModelForClass(Account);
+
+describe('@deepkit/type transforms', () => {
+  let id: string;
+  let accountClassObject: Account;
+  let account: Account;
+  beforeEach(async () => {
+    accountClassObject = new Account();
+    accountClassObject.email = 'somebody@gmail.com';
+    accountClassObject.confidentialProperty = 'secret';
+    account = await AccountModel.create(accountClassObject);
+    // note here that _id is an ObjectId, hence the toString()
+    // otherwise it will have the shape of : { _bsonType: 'ObjectId', id: ArrayBuffer }
+    id = account._id;
+    accountClassObject._id = account._id;
+    accountClassObject.__v = account.__v;
+  });
+
+  describe('lean query', () => {
+    it(`should be able to use @deepkit/type to transform to class instance`, async () => {
+      // lean return a Plain Old Javascript Object
+      const pojo = await AccountModel.findById(id).orFail().lean().exec();
+      // groupsExclude option excludes the property group
+      const access = { groupsExclude: [Group.confidential] };
+      // deserialize pojo back to an Account instance
+      const deserialized = plainToClass(Account, pojo, access);
+      // need to remove the password property, as it is not available due to our @t.group confidential exclusion definition
+      delete accountClassObject.confidentialProperty;
+      expect(deserialized).toStrictEqual(accountClassObject);
+      expect(deserialized).not.toHaveProperty('confidentialProperty');
+    });
+
+    it(`should be able to use @deepkit/type's groups to allow access`, async () => {
+      // lean return a Plain Old Javascript Object
+      const pojo = await AccountModel.findById(id).orFail().lean().exec();
+      // groups option includes the property group
+      const access = { groups: [Group.confidential, Group.public] };
+      // deserialize pojo back to an Account instance
+      const deserialized = plainToClass(Account, pojo, access);
+      expect(deserialized).toStrictEqual(accountClassObject);
+      expect(deserialized).toHaveProperty('confidentialProperty');
+    });
+  });
+
+  describe('normal query', () => {
```

Comment:
is it right not to use `accountClassObject` in this block?

Reply:
Not completely sure what you mean. The test starting around that line is a serialization. It's showing the doc being serialized to a POJO. 

Scott

Reply:
with this question i meant why `accountClassObject` was not used in this test block, if it really is intended, could you move the creation of `accountClassObject` to the block at uses it (not the "global" one)?

Reply:
Ah. Ok. @hasezoey I see what you mean. `accountClassObject` is used twice though and I did change the initialization code  from `beforeAll` to `beforeEach`, so that object is reinstated. It would mean that code would need to go into the two tests to get back to `beforeAll`.  

Scott

Reply:
> It would mean that code would need to go into the two tests to get back to beforeAll.

no it would not, it would just need to be moved from the `describe("@deepkit/type transforms").beforeEach` to the `describe("@deepkit/type transforms").describe("lean query")` block

but now that i actually looked at it again, no need to move it, but original issue stands, why are the `accountClassObject` / `account` values not used in the `describe("@deepkit/type transforms").describe("normal query")` block?
-> actually as a afterthought, this "why not used here" might be related to also using `classToPlain` in the tests instead of `plainToClass` for `normal query`

Reply:
@hasezoey - I'm lost now. 🤪 

My intention with the tests was to show serialization and deserialization. I needed the `accountClassObject` to compare to the deserialized object, i.e. to the POJO that is transformed via `plainToClass`. 

The groups tests are a bit different than with `class-transformer`. With `@deepkit/type`, the logic is, if a group is added, only that group is de-/serialized and all other properties are excluded. That is why I added the two groups and the explanations in the docs. To me, it isn't a super-duper solution. But, it comes down to performance and keeping grouping ugly is just a matter of trying to suggest avoiding it altogether. 

Scott

Reply:
resolved with 1fbe58064faecc415eb50c9f2ea0a27afa5bdc19